### PR TITLE
Receive use_docker flag from the cli

### PR DESCRIPTION
**Pull request by chuchocd85**
*Issue #, if available:*

*Description of changes:*
Add ability to consume the use docker flag from the `cloudformation-cli`.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jotompki — CHANGES_REQUESTED** [review]
Is there a way to prompt for the arguments in the plugins? otherwise each argument addition will require a change to both the cloudformation cli and the plugin. Would be good to structure it in a way to where an argument addition on the plugin just means one change to the plugin.

File: tests/plugin/parser_test.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+def test_setup_subparser_python36():
+    parser = argparse.ArgumentParser()
+    subparsers = parser.add_subparsers(dest="subparser_name")
+
+    sub_parser = setup_subparser_python36(subparsers, [])
+
+    args = sub_parser.parse_args(["-d"])
+
+    assert args.language == "python36"
+    assert args.use_docker is True
```
Comment (jotompki) [inline tests/plugin/parser_test.py:15]: nit: `assert args.use_docker`

File: tests/plugin/parser_test.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+def test_setup_subparser_python37():
+    parser = argparse.ArgumentParser()
+    subparsers = parser.add_subparsers(dest="subparser_name")
+
+    sub_parser = setup_subparser_python37(subparsers, [])
+
+    args = sub_parser.parse_args([])
+
+    assert args.language == "python37"
+    assert args.use_docker is False
```
Comment (jotompki) [inline tests/plugin/parser_test.py:27]: same here can simplify

**Review by jotompki — APPROVED** [review]

**Review by jaymccon — APPROVED** [review]

File: python/rpdk/python/codegen.py [hunk 3]
```diff
@@ -49,23 +49,25 @@ def __init__(self):
...
-        self._use_docker = input_with_validation(
+
+        self._use_docker = self._use_docker or input_with_validation(
             "Use docker for platform-independent packaging (Y/n)?\n",
             validate_no,
             "This is highly recommended unless you are experienced \n"
             "with cross-platform Python packaging.",
         )
+
         project.settings["use_docker"] = self._use_docker
```
Comment (eduardomourar) [inline python/rpdk/python/codegen.py:69]: We should make this `useDocker` for consistency with `protocolVersion` and JSON convention, right?

**Review by ammokhov — APPROVED** [review]

**Pull request merged**